src/models/CourseModel.js

```diff
@@ -0,0 +1,103 @@
+const admin = require('firebase-admin');
+
+class CourseModel {
+  constructor({
+    coursesCode,
+    courseName,
+    typeOfCourse,
+    description,
+    courseImgUrl,
+    logoUrl,
+    points,
+    rating,
+    otherInfo,
+    videoURL = [], // Default to empty array if not provided
```

Comment:
One video per course? Why not make videos part of the lesson object? Similarly, the other fields about quizzes should also live in a Course model.

Reply:
I will discuss with team and will changes course collection fields.